vault/resource_aws_auth_backend_sts_role_test.go

```diff
@@ -20,13 +22,41 @@ func TestAccAWSAuthBackendSTSRole_import(t *testing.T) {
 	backend := acctest.RandomWithPrefix("aws")
 	accountID := strconv.Itoa(acctest.RandInt())
 	arn := acctest.RandomWithPrefix("arn:aws:iam::" + accountID + ":role/test-role")
+
 	resource.Test(t, resource.TestCase{
 		PreCheck:          func() { testutil.TestAccPreCheck(t) },
 		ProviderFactories: providerFactories,
 		CheckDestroy:      testAccCheckAWSAuthBackendSTSRoleDestroy,
 		Steps: []resource.TestStep{
 			{
-				Config: testAccAWSAuthBackendSTSRoleConfig_basic(backend, accountID, arn),
+				Config: testAccAWSAuthBackendSTSRoleConfig_basic(backend, accountID, arn, ""),
+				Check:  testAccAWSAuthBackendSTSRoleCheck_attrs(backend, accountID, arn),
+			},
+			{
+				ResourceName:      "vault_aws_auth_backend_sts_role.role",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccAWSAuthBackendSTSRole_importWithExternalID(t *testing.T) {
```

Comment:
@vinay-gopalan I ended up creating a whole separate import test to test with `external_id`. Lmk if you have any other approach in mind!

Reply:
Writing down some stuff I tried so I can remember when we sync: 

I tried updating the existing `TestAccAWSAuthBackendSTSRole_import` to pass `external_id` and set `ImportStateVerifyIgnore` accordingly, but wasn't able to get both the unit tests and acceptance tests to pass.

Approaches tried:
- Check the version outside the `resource.Test` and set an `importStateVerifyIgnore` slice accordingly. Acceptance tests passed, but I was getting a [nil pointer dereference error](https://github.com/hashicorp/terraform-provider-vault/actions/runs/11925647195/job/33238089474) on `testProvider` in the CI unit tests since I think `testProvider` wasn't getting initialized until later.
- Initialize `importStateVerifyIgnore` slice outside `resource.Test`, then try to assign it based on version in the `PreCheck` func. Unit tests passed (since `testProvider` was initialized by that point), but I don't think `importStateVerifyIgnore` was getting picked up in the final assertions, because we were [still checking for `external_id` on the import acceptance tests in Vault versions < 1.17](https://github.com/hashicorp/terraform-provider-vault/actions/runs/11925787705/job/33238512409).

Reply:
We synced and decided to remove the separate import tests (instead opt for import test steps), and create a separate test for `withExternalID` that only runs on Vault versions >= 1.17. See https://github.com/hashicorp/terraform-provider-vault/pull/2370/commits/67bb7aa2f0b849048741f005c85d535181bfcbf6 for changes